ethereum_clients/ControlledProcess.js

```diff
@@ -40,6 +40,12 @@ class ControlledProcess extends EventEmitter {
       this.state
     )
   }
+  emitPluginError(error) {
+    this.emit('pluginError', {
+      message: error.toString(),
+      key: new Date().getTime() + Math.random()
```

Comment:
were you getting duplicate keys without the Math.random?

Reply:
yes mostly because of the multiple event emitting bug from grid